helix-term/src/application.rs

```diff
@@ -56,15 +56,33 @@ pub struct Application {
 }
 
 impl Application {
-    pub fn new(args: Args, config: Config) -> Result<Self, Error> {
+    pub fn new(args: Args) -> Result<Self, Error> {
```

Comment:
I'm sorry I'm kind of late to the party here, this PR eluded me. This change actually presents a problem. I'm working on an integration testing branch, and this change makes it so that you can't pass in an anyway parsed Config object. This was very useful for testing, as it's super easy to just make the struct literal you want to describe the config option you want to test. Now it seems that the config can *only* come from a toml file. Could we please move this logic back into main?